### DISCUSSION TITLE
Should not destroy() reset zoom and pan settings?

### DISCUSSION THREAD

**Discussion by posterhorn** [chartjs/chartjs-plugin-zoom#703] [Q&A] [answered]
Please see this [JSFiddle](https://jsfiddle.net/wksvzcjn/2/)  for demonstration.

A global ChartConfiguration is used to basically create a chart and enable zooming and panning. 
After zooming manually, destroying the chart and recreating the chart again with reusing the global ChartConfiguration, the scale of the chart is restored to values it had after zooming.

Expected was a recreation of the chart similar to its initial creation (as it is with the second chart using a locally created ChartConfigration shown in the JSFiddle for comparison).

Is not the call of [destroy()](https://www.chartjs.org/docs/latest/developers/api.html#destroy) supposed to already reset the zoom and pan settings in order to leave the global ChartConfiguration unchanged?

**Comment by kurkle** [comment, marked as answer]
You probably need to clean up the config too. I'd suggest a `createConfig` function that always returns a new object.

**Reply by posterhorn** [reply]
Thank you for your answer.

That is an option that definitly works, see [JSFiddle](https://jsfiddle.net/39yuv0qw/).

We were just wondering because the [API description](https://www.chartjs.org/docs/latest/developers/api.html#destroy) says 

> This will clean up any references stored to the chart object within Chart.js, along with any associated event listeners attached by Chart.js.

We probably just misinterpreted that.

**Reply by kurkle** [reply]
Sadly the config object gets modified and you have the reference stored outside chart.js, so it can't clean it up. 
We should look into having an temporary object where the scale options will be stored by zoom plugin, so the original config stays intact.
